ocaml/sdk-gen/java/main.ml

```diff
@@ -1052,6 +1061,29 @@ public class Types
   TypeSet.iter (gen_task_result_func file) !types ;
   fprintf file
     {|
+    
+    public static class BadAsyncResult extends XenAPIException
+    {
+        public final String result;
+
+        public BadAsyncResult(String result)
+        {
+            super(result);
+            this.result = result;
+        }
+    }
+
+    private static String parseResult(String result) throws BadAsyncResult
+    {
+        Pattern pattern = Pattern.compile("<value>(.*)</value>");
+        Matcher matcher = pattern.matcher(result);
+        if (!matcher.find() || matcher.groupCount() != 1) {
+            throw new Types.BadAsyncResult("Can't interpret: " + result);
```

Comment:
If you're making other changes I'd change `interpret` to `parse` as interpretation in SW has a different meaning.